WooCommerce/Classes/ViewRelated/Coupons/Add and Edit Coupons/AddEditCouponViewModel.swift

```diff
@@ -213,6 +213,40 @@ final class AddEditCouponViewModel: ObservableObject {
     }
 
     func updateCoupon(coupon: Coupon) {
+        guard let initialCoupon = self.coupon else {
+            return
+        }
+
+        let usageDetailsUpdated: Bool = {
+            coupon.maximumAmount != initialCoupon.maximumAmount ||
+            coupon.minimumAmount != initialCoupon.minimumAmount ||
+            coupon.usageLimit != initialCoupon.usageLimit ||
+            coupon.usageLimitPerUser != initialCoupon.usageLimitPerUser ||
+            coupon.limitUsageToXItems != initialCoupon.limitUsageToXItems ||
+            coupon.emailRestrictions != initialCoupon.emailRestrictions ||
+            coupon.individualUse != initialCoupon.individualUse ||
+            coupon.excludeSaleItems != initialCoupon.excludeSaleItems ||
+            coupon.excludedProductIds != initialCoupon.excludedProductIds ||
+            coupon.excludedProductCategories != initialCoupon.excludedProductCategories
+        }()
+
+        let expiryDateUpdated: Bool = {
+            guard let oldDate = initialCoupon.dateExpires, let newDate = coupon.dateExpires else {
+                return initialCoupon.dateExpires != coupon.dateExpires
+            }
+            return !oldDate.isSameDay(as: newDate)
+        }()
+
+        ServiceLocator.analytics.track(.couponUpdateInitiated, withProperties: [
```

Comment:
Would it make sense to extract the `couponUpdateInitiated` event tracking code to a private function?

Reply:
Good idea! Updated in d19e7999ee73f7044a8a6d187412d8a8d4aacd6e 👍